### PR TITLE
Some internal refactorings in AnalysisRegistry

### PR DESCRIPTION
**Pull request by cbuescher**
This PR reduces some method scopes and makes a few methods static where they don't use instance fields.
Also slightly changing AnalysisRegistry#produceAnalyze return a NamedAnalyzer instead of having 
a side effect on the analyzer map passed in. In addition, removing the unused `environment`
field from CustomAnalyzerProvider.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java [hunk 1]
```diff
@@ -427,8 +427,14 @@ public IndexAnalyzers build(IndexSettings indexSettings,
         Map<String, NamedAnalyzer> normalizers = new HashMap<>();
         Map<String, NamedAnalyzer> whitespaceNormalizers = new HashMap<>();
         for (Map.Entry<String, AnalyzerProvider<?>> entry : analyzerProviders.entrySet()) {
-            processAnalyzerFactory(indexSettings, entry.getKey(), entry.getValue(), analyzers,
-                tokenFilterFactoryFactories, charFilterFactoryFactories, tokenizerFactoryFactories);
+            analyzers.merge(entry.getKey(), produceAnalyzer(entry.getKey(), entry.getValue(), tokenFilterFactoryFactories,
+                    charFilterFactoryFactories, tokenizerFactoryFactories), (k, v) -> {
+                        throw new IllegalStateException("already registered analyzer with name: " + entry.getKey());
+                    });
+            final String analyzerAliasKey = "index.analysis.analyzer." + entry.getKey() + ".alias";
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java]: The original code had this comment: // TODO: remove alias support completely when we no longer support pre 5.0 indices
Should we remove this part as we don't support pre 5.0 indexes any more

**Review by mayya-sharipova — APPROVED** [review]
@cbuescher  Thanks, very nice refactoring indeed. Side effects are difficult to track, and returning a value instead is much better. I wonder if we should do the same about `processNormalizerFactory`.

File: server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java (hunk 1 above)
Comment (cbuescher) [reply server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java]: Thanks, I was wondering the same. I understood the comment to mean we want to remove the alias support (which I think we have already) but not the error that prevents these settings to be silently ignored going forward. But I will take another look and maybe consult with sb. else before proceeding to merge this.

**Comment by cbuescher** [comment]
> I wonder if we should do the same about `processNormalizerFactory`.

Good point, I now saw they look quite similar, I didn't touch that method in my own refactorings so I missed it. I'd like to do that in a follow up though if possible.

File: server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java (hunk 1 above)
Comment (jimczi) [reply server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java]: +1 to remove, this is not supported nor documented since version 5.

**Pull request merged**